kubejs/server_scripts/base/featrues/bulkrecycle-command.js

```diff
@@ -0,0 +1,166 @@
+/**
+ * 
+ * @param {Internal.ServerPlayer} player 
+ */
+function compactItems(player,keepOne){
+
+  let limit = keepOne ? 1 : 0;
+
+  let tallies = {}
+  player.inventory.allItems.toArray().forEach(
+    /** @param {Internal.ItemStack} i */
+    (i,slot)=>{
+      if(i.hasTag("forge:devices")){
+        //highly esoteric way to do it, so sorry!
+        let [type,fraction] = i.tags.toArray().join().match(/devices\/([^\]\s]+)]/)[1].split("/")
```

Comment:
You should be able to do this without regex, I have some code for that in the new tooltips script, something like this 
```js
  let deviceTag = item2
    .getTags()
    .toList()
    .filter((tag) => tag.location().path.startsWith("device/"))[0];
```
so you can probably do
```js
let deviceSuffix = deviceTag.location().path.split("/")[2];
```
or similar